go/private/actions/archive.bzl

```diff
@@ -52,6 +52,11 @@ def emit_archive(go, source = None):
     split = split_srcs(source.srcs)
     lib_name = source.library.importmap + ".a"
     out_lib = go.declare_file(go, path = lib_name)
+    out_archives = [out_lib]
+    out_linklib = None
+    if go.mode.linkobj:
+        out_linklib = go.declare_file(go, path = source.library.importmap + ".link.a")
```

Comment:
Ideally, we should store export data in the .x file instead of in a new file. Currently, the .a file contains export data and compiled code, and the .x file contains facts. We should make it so both .a and .x files are archives, and .a contains compiled code, and .x contains export data and facts. 

With that change, we wouldn't need to change the toolchain or provider API at all. However, nogo would need to be able to read facts from a file within an archive instead of reading the whole file (we already have code for parsing archives, so probably not too hard).

Reply:
I don't see that `.x` file being part of the provider? You'd rather add a field?
Because that `.x` file needs to be passed to `compilepkg` and only the compiled code to the linker.

Reply:
The .x file is already exposed as `GoArchive.data.export_file`. It doesn't look like it's documented yet, but it probably should be.